src/main/java/com/bigstep/datalake/DLFileSystem.java

```diff
@@ -1661,4 +1795,174 @@ protected HttpURLConnection connect(final long offset,
             return new URLRunner(GetOpParam.Op.OPEN, offsetUrl, resolved).run();
         }
     }
+
+    private static class DLEncryptionUtils {
+        private static String strName = "DLENCRYPTION";
+        private static String strVersion = "1.0";
+        private static String strCipher = "AES/CTR/NoPadding";
+        private final static int nHeaderDetailLength = 128;
+        private final static int nIVLength = 16;
+        private final static int nAesKeyLength = 16;
+        private static boolean _isInitialised = false;
```

Comment:
I would suggest using a singleton rather than using all this static stuff. The reason is that it will be harder to to test  this class, to replace it with another implementation at a later time, or rather simply derive a different class from it later.